### ISSUE TITLE
can't be used from TypeScript

### ISSUE THREAD

**Issue by RohanHart** [then/lazy-promise#4]
The index.d.ts provided has insufficient details - particularly no constructor.
The index.d.ts generated by tsc seems to be correct so there should be no need to override.

**Comment by ForbesLindesay** [comment]
I don't know what index.d.ts you are referring to. We auto-generate one via tsc, but we don't manually define one anywhere in this module.

**Comment by RohanHart** [comment]
The last line of prepare.js overwrites the index.d.ts
``writeFileSync(__dirname + '/lib/index.d.ts', 'export default Promise;');

**Comment by ForbesLindesay** [comment]
Ahhh, that makes sense. What's your tsconfig?  I think we might be assuming you have es2016 enabled.

I think we don't accept the default config since it made it difficult to pass a `LazyPromise` to a function that expects a Promise.

**Comment by RohanHart** [comment]
Yes, targeting es2015. I've created a local typing which says that LazyPromise just implements PromiseLike as that's enough for what I need.
